modules/4337/test/e2e/run.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+docker compose up -d
+# wait for containers to start up
+SECONDS=0
+until curl -fs http://localhost:8545 >/dev/null && curl -fs http://localhost:3000 >/dev/null; do
+    if [[ $SECONDS -gt 30 ]]; then
+        echo "ERROR: timeout waiting for local node and bundler to start"
+        docker compose logs
+        exit 1
+    fi
+    sleep 1
+done
+
+hardhat test --deploy-fixture --network localhost --grep '^E2E - '
+
+docker compose down
```

Comment:
Interesting that you do a tear-down for every test run. I hadn't really thought about it, but it makes it so things run from a more deterministic state :+1:.

Reply:
Neither had I 😄